src/block/index.js

```diff
@@ -10,6 +10,10 @@ module.exports = (bridgeState, db, nodeConfig = {}, sender) => async (
   chainInfo
 ) => {
   bridgeState.checkCallsCount = 0;
+
+  // delete collected votes for submitted period
+  delete (state.periodVotes || {})[bridgeState.lastBlocksRoot];
```

Comment:
i don't understand how this can work, if you delete it every block. shouldn't it be every period (32 blocks) ?

Reply:
`bridgeState.lastBlocksRoot` can contain root for submitted period only (it is ready from Submission event). Submitted period = we don't need the votes for it anymore. If there are not enough votes yet, then the period won't be submitted (subject to be done in #309) and hence nothing will be deleted here